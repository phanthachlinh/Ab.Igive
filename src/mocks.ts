import { ITranslationsDictionary } from "./types/general";

export const TEST_TRANSLATIONS: ITranslationsDictionary = {
	en: {
		homepage: {
			additionalEventListItem: {
				title: 'Still not sure what to do or how to do it?',
				subtitle: 'Get in touch. We’ll be happy to help you define your campaign and set up your fundraising page.',
				buttonText: 'CONTACT US',
				buttonLink: 'https://mycause.unhcr.org/contact_forms/new'
			}
		},
		eventPage: {
			title: 'Start your campaign today.',
			subTitle: 'It only takes two minutes to set up your fundraising page. Decide what to do and choose in which global campaign you want to get involved in. Pick a name, a photo and just like that, you’ll be ready to start raising money to help people who have been forced to flee.'
		},
		shared: {
			seeAllCampaigns: 'see all campaigns',
			searchForGlobalCamp: 'Search for a global campaign',
			copyLink: 'Copy link',
			donateNow: 'DONATE NOW',
			shareThisCampaign: ' Share this campaign',
			setupMyFundpage: 'Setup my fundraising page',
			shareThisFundPage: 'Share this fundraising page',
			createMyFundPage: 'Create my fundraising page'
		}
	},
	hk: {
		homepage: {
			additionalEventListItem: {
				title: '還未肯定應該做甚麼？',
				subtitle: '與我們聯繫吧。我們樂意幫你為籌款項目找到定位，以及協助你創建籌款專頁。',
				buttonText: '聯絡我們',
				buttonLink: 'https://mycause.unhcr.org/contact_forms/new'
			}
		},
		eventPage: {
			title: '今日就建立你的籌款專頁',
			subTitle: '創建你的籌款專頁只需兩分鐘。先決定你想開設的項目類型並選擇其中一個全球項目。為活動起一個名字以及挑選一張相片，就這麼簡單，你就準備好建立籌款專頁，協助被迫逃離家園的難民。'
		},
		shared: {
			seeAllCampaigns: '查看所有籌款活動',
			searchForGlobalCamp: '搜尋全球項目',
			copyLink: '複製連結',
			donateNow: '立即捐款',
			shareThisCampaign: '分享此籌款活動',
			setupMyFundpage: '設置我的籌款頁面',
			shareThisFundPage: '分享此籌款專頁',
			createMyFundPage: '建立我的籌款頁面'
		}
	}
}
