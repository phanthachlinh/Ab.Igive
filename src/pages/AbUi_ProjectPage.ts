/**
* Class renders changes on the project page
**/
import { ITranslationsDictionary } from "../types/general";
import AbUi_ShareLink from "../components/AbUi_ShareLink";
export default class AbUi_ProjectPage {

	translations: ITranslationsDictionary;
	language: string;
	AbUiShareLink: AbUi_ShareLink
	projectInfoContainer: HTMLDivElement | null;

	constructor(translations: ITranslationsDictionary, language: string) {

		this.translations = translations
		this.language = language
		this.AbUiShareLink = new AbUi_ShareLink(this.translations, this.language)
		this.projectInfoContainer = document.querySelector('.p2p-detail-data>div:first-child')
	}
	/**
	* renders the share link component
	**/
	private _renderShareLinkComponent(): void {
		if (!this.projectInfoContainer) {
			console.error('Div project info container was not found');
			return
		}
		let shareLinkHtmlString: string = this.AbUiShareLink.getShareLinkComponent();
		this.projectInfoContainer.insertAdjacentHTML('beforeend', shareLinkHtmlString);
		this.AbUiShareLink.attachEventListener()

	}

	/**
	* changes the video height
	**/
	private _adaptVideoHeight() {
		if (window.innerWidth >= 768) {
			let height = document.querySelector('.project-detail-data') ?.clientHeight
      let videoContainers: Array<HTMLUListElement> = Array.from(document.querySelectorAll('.bxslider .maybevideocontainer'))
			videoContainers.map((el: HTMLUListElement) => {
				el.style.height = height + 'px'
			});
		}
	}

	render() {
		this._renderShareLinkComponent()
		this._adaptVideoHeight()
	}
}
