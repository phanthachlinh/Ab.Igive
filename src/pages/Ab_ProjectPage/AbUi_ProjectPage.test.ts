import AbUi_ProjectPage from './AbUi_ProjectPage'
import { TEST_TRANSLATIONS } from '../../mocks'
import { JSDOM } from 'jsdom'
declare var global: any
let abUiProjectPageObj: any
console.error = jest.fn((errmsg: string) => {
	throw errmsg
})
describe('AbUi_ProjectPage', () => {
	describe('_renderShareLinkComponent', () => {
		it('should throw projectInfoContainer not defined', () => {
			global.document.documentElement.innerHTML = new JSDOM(``)
			abUiProjectPageObj = new AbUi_ProjectPage(TEST_TRANSLATIONS, 'en')
			expect(() => { abUiProjectPageObj._renderShareLinkComponent() }).toThrow('Div project info container was not found')
		})
		it('should match snapshot', () => {
			global.document.documentElement.innerHTML = new JSDOM(`<div class="p2p-detail-data"><div></divu></div>`).serialize()
			abUiProjectPageObj = new AbUi_ProjectPage(TEST_TRANSLATIONS, 'en')
			abUiProjectPageObj._renderShareLinkComponent()
			expect(document.documentElement.innerHTML).toMatchSnapshot()
		})
	})
	describe('_adaptVideoHeight', () => {

	})
})
