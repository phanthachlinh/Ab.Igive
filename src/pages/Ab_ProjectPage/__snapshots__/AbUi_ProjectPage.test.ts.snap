// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbUi_ProjectPage _renderShareLinkComponent should match snapshot 1`] = `
"<head></head><body><div class=\\"p2p-detail-data\\"><div>
      <div class=\\"share-link\\">
        <h6> Share this campaign</h6>
        <div class=\\"share-link__wrapper\\">
        <input type=\\"text\\" value=\\"http://localhost/\\">
        <button>Copy link</button>
        </div>
      </div>
    </div></div></body>"
`;
